### DISCUSSION TITLE
Opensaber to sunbird-rc-core migration

### DISCUSSION THREAD

**Discussion by maheshkumargangula** [Sunbird-RC/community#125] [Issues]
@dileepbapat - We are using the Openseber version of the sunbird registry. With sunbird-RC-core, how to use `compatibility mode` configuration and deploy the latest version to support backward compatibility.

We would like to upgrade to the latest version of the sunbird registry. Please let us know the steps to do it.

Cc: @vrayulu

**Comment by pramodkvarma** [comment]
I am curious @maheshkumargangula, just replacing the current instance of opensaber with latest Sunbird RC instance is an issue? What's breaking? Can you describe?

Also adding @tejash-jl to thread in case @dileepbapat is tied up.

**Reply by maheshkumargangula** [reply]
@tejash-jl @dileepbapat - We are using [open-saber](https://github.com/Sunbird-RC/open-saber) for the `Vidyadaan` project. `Vidyadaan` uses sunbird `user-service` for authentication and open-saber is only for storing the contributor details. We want to start with the same state and then integrate Keycloak with sunbird RC in a phased manner.

Replacing the current instance of [open-saber](https://github.com/Sunbird-RC/open-saber) with the latest Sunbird RC is having a backward compatibility issue with the APIs. 

To create an object (ex: User) we use `POST <basepath>/add` API. This API working fine with open-saber (It is not expecting any JWT token). But with the latest Sunbird RC, the same API endpoint not responding. By checking the service logs, it looks like redirecting to Keycloak for login and the JWT token.

```log
12:27:06.137 [http-nio-8081-exec-1] DEBUG o.k.adapters.PreAuthActionsHandler:74 - adminRequest http://localhost:8081/add
12:27:06.140 [http-nio-8081-exec-1] DEBUG o.k.a.t.AbstractAuthenticatedActionsValve:57 - AuthenticatedActionsValve.invoke /add
12:27:06.141 [http-nio-8081-exec-1] DEBUG o.k.a.AuthenticatedActionsHandler:53 - AuthenticatedActionsValve.invoke http://localhost:8081/add
12:27:06.141 [http-nio-8081-exec-1] DEBUG o.k.a.AuthenticatedActionsHandler:153 - Policy enforcement is disabled.
12:27:06.143 [http-nio-8081-exec-1] DEBUG o.k.adapters.PreAuthActionsHandler:74 - adminRequest http://localhost:8081/add
12:27:06.157 [http-nio-8081-exec-1] DEBUG o.k.a.s.a.KeycloakAuthenticationEntryPoint:116 - Redirecting to login URI /sso/login
12:27:06.161 [http-nio-8081-exec-2] DEBUG o.k.adapters.PreAuthActionsHandler:74 - adminRequest http://localhost:8081/sso/login
12:27:06.161 [http-nio-8081-exec-2] DEBUG o.k.a.t.AbstractAuthenticatedActionsValve:57 - AuthenticatedActionsValve.invoke /sso/login
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.AuthenticatedActionsHandler:53 - AuthenticatedActionsValve.invoke http://localhost:8081/sso/login
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.AuthenticatedActionsHandler:153 - Policy enforcement is disabled.
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.adapters.PreAuthActionsHandler:74 - adminRequest http://localhost:8081/sso/login
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.s.f.KeycloakAuthenticationProcessingFilter:206 - Request is to process authentication
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.s.f.KeycloakAuthenticationProcessingFilter:142 - Attempting Keycloak authentication
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.s.t.SpringSecurityTokenStore:64 - Checking if org.keycloak.adapters.springsecurity.authentication.SpringSecurityRequestAuthenticator@6f272053 is cached
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.OAuthRequestAuthenticator:266 - there was no code
12:27:06.162 [http-nio-8081-exec-2] DEBUG o.k.a.OAuthRequestAuthenticator:274 - redirecting to auth server
12:27:06.163 [http-nio-8081-exec-2] DEBUG o.k.a.OAuthRequestAuthenticator:144 - callback uri: http://localhost:8081/sso/login
12:27:06.163 [http-nio-8081-exec-2] DEBUG o.k.a.s.f.KeycloakAuthenticationProcessingFilter:155 - Auth outcome: NOT_ATTEMPTED
12:27:06.163 [http-nio-8081-exec-2] DEBUG o.k.a.OAuthRequestAuthenticator:227 - Sending redirect to login page: http://localhost:8080/auth/realms/sunbird-rc/protocol/openid-connect/auth?response_type=code&client_id=registry-frontend&redirect_uri=http%3A%2F%2Flocalhost%3A8081%2Fsso%2Flogin&state=bb8f0a8d-db1d-4ba1-9e2f-8f83c3fd415e&login=true&scope=openid
```

We checked only this API so far. The latest sunbird RC service is running and the Health API also working fine.

Cc: @pramodkvarma

**Comment by dileepbapat** [comment]
ok, so you are running with authentication.enabled=false?  Probably interceptor is applied for legacy apis also, I will take a look and update if that is the case.

**Reply by maheshkumargangula** [reply]
yes @dileepbapat `authentication.enabled=false`. That is the default configuration we have.
